python/ray/util/data/dataset.py

```diff
@@ -0,0 +1,359 @@
+import random
+from dataclasses import is_dataclass
+from functools import wraps
+from typing import Callable, List, Union, Iterable, Iterator
+
+import pandas as pd
+from pandas import DataFrame
+
+from ray.util.iter import (_ActorSet, _NextValueNotReady, LocalIterator, ParallelIterator,
+                           T, U)
+
+
+class MLDataset(ParallelIterator[DataFrame]):
```

Comment:
Shall we use `pd.DataFrame` consistently to better disambiguate vs other dataframes?

Reply:
updated